ocp_resources/virtual_machine_clone.py

```diff
@@ -0,0 +1,82 @@
+from ocp_resources.constants import TIMEOUT_4MINUTES
+from ocp_resources.resource import NamespacedResource
+from ocp_resources.utils import TimeoutSampler
+from ocp_resources.virtual_machine import VirtualMachine
+
+
+class VirtualMachineClone(NamespacedResource):
+    api_group = NamespacedResource.ApiGroup.CLONE_KUBEVIRT_IO
+
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        source_name=None,
+        source_type=None,
+        target_name=None,
+        label_filters=None,
+        annotation_filters=None,
+        new_mac_addresses=None,
+        new_smbios_serial=None,
+        yaml_file=None,
+        **kwargs,
+    ):
+        """
+        Create VirtualMachineClone object.
+
+        Args:
+            name (str): VirtualMachineClone name
+            source_name (str): the clone's source name
+            source_type (str): the clone's source type
+            target_name (str): the clone's target name
+            label_filters (list, optional): List of label filters, e.g. ["*", "!someKey/*"]
+            annotation_filters (list, optional): List of annotation filters, e.g. ["firstKey/*", "secondKey/*"]
+            new_mac_addresses (dict, optional): Dict of new MAC addresses, {interface_name: mac_address}
+            new_smbios_serial (str, optional): the clone's new smbios serial
+        """
+        super().__init__(**kwargs)
+        self.source_name = source_name
+        self.source_type = source_type
+        self.target_name = target_name
+        self.label_filters = label_filters
+        self.annotation_filters = annotation_filters
+        self.new_mac_addresses = new_mac_addresses
+        self.new_smbios_serial = new_smbios_serial
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            spec = self.res.setdefault("spec", {})
+            source = spec.setdefault("source", {})
+            target = spec.setdefault("target", {})
+
+            source["apiGroup"] = NamespacedResource.ApiGroup.KUBEVIRT_IO
+            source["kind"] = self.source_type
+            source["name"] = self.source_name
+
+            target["apiGroup"] = NamespacedResource.ApiGroup.KUBEVIRT_IO
+            target["kind"] = VirtualMachine.kind
+            target["name"] = self.target_name
+
+            if self.label_filters:
+                spec["labelFilters"] = self.label_filters
+            if self.annotation_filters:
+                spec["annotationFilters"] = self.annotation_filters
+            if self.new_mac_addresses:
+                spec["newMacAddresses"] = self.new_mac_addresses
+            if self.new_smbios_serial:
+                spec["newSMBiosSerial"] = self.new_smbios_serial
+
+    def wait_complete(self):
+        self.logger.info(
```

Comment:
Any reason to not use`wait_for_status()` ?

Reply:
thanks, removed this part since `wait_for_status()` does all the job